src/less-cache.coffee

```diff
@@ -194,15 +168,37 @@ class LessCache
       @writeJson(@getCachePath(@importsFallbackDir, filePath), cacheEntry)
 
   parseLess: (filePath, contents) ->
+    entryPath = filePath.replace(/[^\/\\]*$/, '')
+
+    # https://github.com/less/less.js/blob/ef4baa5bb27b932623eb9afede99b3e2aaccea20/packages/less/src/less/contexts.js#L18
+    options = {syncImport: true, paths: @importPaths}
+    context = new less.contexts.Parse(options)
+
+    # https://github.com/less/less.js/blob/ef4baa5bb27b932623eb9afede99b3e2aaccea20/packages/less/src/less/import-manager.js#L9
+    rootFileInfo = {
+      filename: filePath,
+      rootpath: '',
+      currentDirectory: entryPath,
+      entryPath: entryPath,
+      rootFilename: filePath
+    }
+    importManager = new less.ImportManager(less, context, rootFileInfo)
+
     css = null
-    options = filename: filePath, syncImport: true, paths: @importPaths
-    less ?= require('less')
-    imports = @observeImportedFilePaths ->
-      less.render contents, options, (error, result) ->
-        if error?
-          throw error
-        else
-          {css} = result
+    parser = new less.Parser(context, importManager, rootFileInfo)
+
+    parser.parse contents, (err, rootNode) ->
+      if err?
+        throw error
+      else
+        parserTree = new less.ParseTree(rootNode, importManager)
+        {css} = parserTree.toCSS(options)
```

Comment:
@matthew-dean do you why this does not work? The `css` variable is null most of the time.
You can run the tests by `npm run test`, and put `console.log` to check this.